### PR TITLE
[MM-24591] Improve simulcontroller stop consistency

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
With this PR, 
- `simulcontroller`: make `connect` and `disconnect` functions communicate through channels. So that we can be sure about all goroutines and channels are clean before logging out.
- `user`: remove `Cleanup()` method.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-24591

### CODE REVIEW

File: loadtest/control/simulcontroller/actions.go [hunk 1]
```diff
@@ -28,18 +28,37 @@ func (c *SimulController) connect() error {
...
 	}()
+	go c.wsEventHandler()
+	go c.periodicActions()
+	return nil
+}
 
+func (c *SimulController) disconnect() error {
+	// one for ws loop and one for periodic actions loop
+	select {
+	case <-c.connected:
```
Comment (agnivade) [inline loadtest/control/simulcontroller/actions.go:45]: Same as above, we should do a CAS and then move on. This is un-idiomatic and error-prone. What happens if one calls connect consecutively twice ? Or even calls disconnect twice ? :)

Either we use CAS, or not. 0/5 on that. But we shouldn't use buffered channels to track state.

File: loadtest/user/userentity/user.go [hunk 2]
```diff
@@ -100,6 +99,8 @@ func (ue *UserEntity) Connect() (<-chan error, error) {
 		return nil, errors.New("user is already connected")
 	}
 
+	ue.wsEventChan = make(chan *model.WebSocketEvent, 1)
```
[the remark this answers is not in the document]
Comment (isacikgoz) [reply loadtest/user/userentity/user.go]: Resolved offline, will wait to disconnect first.

File: loadtest/control/simulcontroller/actions.go [hunk 3]
```diff
@@ -25,22 +26,45 @@ type userAction struct {
...
 	}
+	c.wg.Add(3)
 	go func() {
+		defer c.wg.Done()
 		for err := range errChan {
 			c.status <- c.newErrorStatus(err)
 		}
 	}()
+	go c.wsEventHandler(c.wg)
+	go c.periodicActions(c.wg)
```
Comment (streamer45) [inline loadtest/control/simulcontroller/actions.go:45]: Any reason we are passing the waitgroup? We have it as part of the controller after all.

Comment (isacikgoz) [reply loadtest/control/simulcontroller/actions.go:45]: Just to make more explicit where `wg` is incremented and consumed.

Comment (streamer45) [reply loadtest/control/simulcontroller/actions.go:45]: Fair enough :+1:

**Comment by streamer45** [comment]
I am glad to say this has passed one of my most gruesome load-tests to date without any race report nor sign of deadlock. Happy to merge it :+1:

**Pull request merged**